internal/builders/bazel/build.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -eu
+#
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+mkdir binaries
+
+# Transfer flags and targets to their respective arrays
+IFS=' ' read -r -a BUILD_FLAGS <<< "${FLAGS}"
+IFS=' ' read -r -a BUILD_TARGETS <<< "${TARGETS}"
+
+# Build with respect to entire arrays of flags and targets
+bazel build "${BUILD_FLAGS[@]}" "${BUILD_TARGETS[@]}"
+
+# Using target string, copy artifact to binaries dir
+for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
+  CD_PATH=${CURR_TARGET%:*}
+  CD_PATH=${CD_PATH////}
+  BINARY_NAME=${CURR_TARGET#*:}
```

Comment:
I know we're not always good about it but bash is often difficult to understand/remember. Consider adding a comment as to what each of these lines are doing.

Reply:
Definitely, I hear you and agree! Will add comments!

Reply:
Comments added on new bash script.